CMakeLists.txt

```diff
@@ -0,0 +1,630 @@
+            set(LINK_FLAGS_RELEASE  "${LINK_FLAGS_RELEASE} -Wl,--gc-sections -Wl,-s")
+        else()
+            set(LINK_FLAGS_RELEASE  "${LINK_FLAGS_RELEASE} -dead_strip")
+        endif()
+    endif()
+endif()
+
+# Global optimization flags
+if(NOT MSVC)
+    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -fno-omit-frame-pointer")
+endif()
+
+string(TOLOWER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE_LOWER)
+
+if(CMAKE_BUILD_TYPE_LOWER STREQUAL "release")
+    add_definitions(-DNDEBUG)
+endif()
+
+if(MSVC)
+    # Force to always compile with W4
+    if(CMAKE_C_FLAGS MATCHES "/W[0-4]")
+        string(REGEX REPLACE "/W[0-4]" "/W4" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+    else()
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /W4")
+    endif()
+elseif(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
+    # Update if necessary
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -pedantic")
+endif()
+
+# Disable bogus MSVC warnings
+if(MSVC)
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+endif()
+
+# -fPIC thing
+if(NOT WIN32)
+    option(LIBXMP_PIC   "Build library as the position-independent code" OFF)
+    if(LIBXMP_PIC)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+    endif()
+endif()
+
+# Prevent shared libraries has "lib" prefix on Windows DLL files
+if(WIN32)
+    set(CMAKE_SHARED_LIBRARY_PREFIX "")
+endif()
+# -----------------------------------
+
+
+if(POLICY CMP0077) # Cache variables override since 3.12
+    cmake_policy(SET CMP0077 NEW)
+endif()
+
+if(EMSCRIPTEN)
+    set(BUILD_SHARED_DEFAULT OFF)
+else()
+    set(BUILD_SHARED_DEFAULT ON)
+endif()
+
+option(BUILD_STATIC   "Build static library of libXMP" ON)
+option(BUILD_SHARED   "Build shared library of libXMP" ${BUILD_SHARED_DEFAULT})
+if(NOT BUILD_STATIC AND NOT BUILD_SHARED)
+    message(FATAL_ERROR "Both static and shared builds got disabled. You must enable at least one of them.")
+endif()
+
+option(LIBXMP_BUILD_LITE   "Build lite version of libXMP" ON)
+option(LIBXMP_BUILD_FULL   "Build full version of libXMP" ON)
+if(NOT LIBXMP_BUILD_LITE AND NOT LIBXMP_BUILD_FULL)
+    message(FATAL_ERROR "Both Full and Lite versions got disabled. You must enable at least one of them.")
+endif()
+
+option(LIBXMP_DISABLE_IT   "Disable IT support at libXMP-Lite" OFF)
+option(LIBXMP_DISABLE_DEPACKERS     "Disable archive depackers at libXMP" OFF)
+option(LIBXMP_DISABLE_PROWIZARD     "Disable the ProWizard at libXMP" OFF)
+
+TEST_BIG_ENDIAN(WORDS_BIGENDIAN)
+if(WORDS_BIGENDIAN)
+    add_definitions(-DWORDS_BIGENDIAN=1)
+endif()
+
+if(CMAKE_BUILD_TYPE_LOWER STREQUAL "debug")
+    add_definitions(-DDEBUG -D_DEBUG -DDEBUG_NOCOLOR)
+endif()
+
+check_include_file(alloca.h HAVE_ALLOCA_H)
+if(HAVE_ALLOCA_H)
+    add_definitions(-DHAVE_ALLOCA_H=1)
+endif()
+
+check_symbol_exists(round "math.h;tgmath.h" HAVE_ROUND_FUNCTION)
+if(HAVE_ROUND_FUNCTION)
+    add_definitions(-DHAVE_ROUND=1)
+endif()
+
+check_symbol_exists(powf "math.h;tgmath.h" HAVE_POWF_FUNCTION)
```

Comment:
Is checking in tgmath.h really necessary?

Reply:
Yes, otherwise on macOS and on somewhere the check gets failed as I reminded, I made this check a while ago...

Reply:
I looked at Mac OS X SDKs 10.4, 10.6. 10.8, 10.13 and 11.3,
and `powf` is always prototyped in math.h.  Can you tell me in
which SDK specifically it fails without tgmath.h? (And remember
that we don't ever use tgmath.h in libxmp.=

Reply:
Ok, that's fine, will remove tgmath.